src/training-data.cc

```diff
@@ -211,6 +225,7 @@ NAN_METHOD(TrainingData::setTrainData) {
 	}
 	TrainingData *self = Nan::ObjectWrap::Unwrap<TrainingData>(info.Holder());
 	self->trainingData->set_train_data(dataSetLength, numInputNodes, &inputVector[0], numOutputNodes, &outputVector[0]);
+	Nan::AdjustExternalMemory(fanny::getTrainingDataByteCount(self));
```

Comment:
You need to get the before and after memory count here as well, like with merge().  If the training data isn't empty, set_train_data will free it.